Code/Framework/AzFramework/AzFramework/Archive/NestedArchive.h

```diff
@@ -68,6 +68,9 @@ namespace AZ::IO
         // deletes all files from the archive
         int RemoveAll() override;
 
+        // lists all the files in the archive
+        int ListAllFiles(AZStd::vector<AZStd::string>& fileEntries) override;
```

Comment:
We should have this return an `AZStd::vector<AZ::IO::Path>` instead of `AZStd::vector<AZStd::string>`

Reply:
Maybe.

I wanted to follow the pre-existing pattern of other `NestedArchive` functions returning int `ZD_*` error codes, hence the out param.

For the change to `AZ::IO::Path`, I think we could do that later.  Introducing it now could touch a lot of places.  There's many `AZStd::string` for paths in ArchiveComponent API right now, and they would be good to update to Paths in one update.

Reply:
By return I didn't mean actually change the return value to an `int`
But have the function accept an `AZStd::vector<AZ::IO::Path>&` instead of an `AZStd::vector<AZStd::string>&`

Reply:
I've changed it to `AZStd::vector<AZ::IO::Path>` on the `INestedArchive` interface, but stopped short of changing it on the `ArchiveComponent` interface.  I tried it out and saw, as predicted, it would force more changes through AssetBundler that I don't want to make right now.